nextcord/interactions/message_component.py

```diff
@@ -1,102 +1,44 @@
+
+from .. import utils
+from ..embeds import Embed
+from ..enums import InteractionResponseType, InteractionType
+from ..errors import ClientException, HTTPException, InteractionResponded, InvalidArgument
+from ..file import File
+from ..flags import MessageFlags
+from ..message import Attachment, Message
+from ..webhook.async_ import WebhookMessage, async_context, handle_message_parameters
+from .base import (
+    MISSING,
+    Interaction,
+    InteractionMessage,
+    InteractionResponse,
+    PartialInteractionMessage,
+    _InteractionMessageState,
 )
 
-if TYPE_CHECKING:
-    from aiohttp import ClientSession
-
-    from .application_command import BaseApplicationCommand, SlashApplicationSubcommand
-    from .channel import CategoryChannel, ForumChannel, StageChannel, TextChannel, VoiceChannel
-    from .client import Client
-    from .guild import Guild
-    from .message import AllowedMentions
-    from .state import ConnectionState
-    from .threads import Thread
-    from .types.interactions import Interaction as InteractionPayload, InteractionData
-    from .ui.modal import Modal
-    from .ui.view import View
-
-    InteractionChannel = Union[
-        VoiceChannel,
-        StageChannel,
-        TextChannel,
-        CategoryChannel,
-        Thread,
-        PartialMessageable,
-        ForumChannel,
-    ]
-
-MISSING: Any = utils.MISSING
-
-ClientT = TypeVar("ClientT", bound="Client")
-
-
-class InteractionAttached(dict):
-    """Represents the attached data of an interaction.
-
-    This is used to store information about an :class:`Interaction`. This is useful if you want to save some data from a :meth:`ApplicationCommand.application_command_before_invoke` to use later in the callback.
-
-    Example
-    -------
+__all__ = ("MessageComponentInteraction",)
 
-    .. code-block:: python3
-
-        async def attach_db(interaction: Interaction):
-            interaction.attached.db = some_real_database()
-
-        async def release_db(interaction: Interaction):
-            interaction.attached.db.close()
-
-        @bot.slash_command()
-        @application_checks.before_invoke(attach_db)
-        @application_checks.after_invoke(release_db)
-        async def who(interaction: Interaction): # Output: <User> used who at <Time>
-            data = interaction.attached.db.get_data(interaction.user.id)
-            await interaction.response.send_message(data)
-    """
-
-    def __init__(self) -> None:
-        super().__init__()
-        self.__dict__ = self
+if TYPE_CHECKING:
+    from ..message import AllowedMentions
+    from ..state import ConnectionState
+    from ..types.interactions import MessageComponentInteraction as MessageComponentPayload
+    from ..ui.modal import Modal
+    from ..ui.view import View
+    from .modal_submit import ModalSubmitInteraction
 
-    def __repr__(self) -> str:
-        return f"<InteractionAttached {super().__repr__()}>"
 
+class MessageComponentInteraction(Interaction):
```

Comment:
A lot of this code is duplicated across both classes in application.py. Move into a seperate super class or base.py